### PR TITLE
[internal review] Add script for issue 1883

### PR DESCRIPTION
**Pull request by VNikiforov-dev**
ATF Test Scripts to check #[_1883]

This PR **ready** for review.
### Summary

New ATF scripts to check the issue.
### ATF version

develop
### Changelog

Update old script for the issue 1883
Extend coverage for issue 1883
### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/Defects/8_1/1883_1_ChoiceSet_being_processed_in_DeleteInteractionChoceSet.lua [hunk 1]
```diff
@@ -0,0 +1,103 @@
+---------------------------------------------------------------------------------------------------
+-- Issue: https://github.com/SmartDeviceLink/sdl_core/issues/1883
+---------------------------------------------------------------------------------------------------
+-- Description: ChoiceSet being processed in DeleteInteractionChoceSet
```
Comment (dboltovskyi) [inline test_scripts/Defects/8_1/1883_1_ChoiceSet_being_processed_in_DeleteInteractionChoceSet.lua:4]: I would rewrite description in a format of requirement.
E.g.: 
> Check SDL rejects PerfromInteraction request if it's being sent during deleting of InteractionChoiceSet

File: test_scripts/Defects/8_1/1883_2_ChoiceSet_being_processed_in_DeleteInteractionChoceSet_after_resumption.lua [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+-- 2. Mobile app is registered and activated
+-- In case:
+-- 1. App is created InteractionChoiseSet(choiceSetID_1)
+-- 2. Unexpected disconnect and reconnect are performed
+-- 3. App re-registered with actual hashId
+-- 4. App starts deleted InteractionChoiseSet(choiceSetID_1)
+-- - During HMI processes VR.DeleteCommands_request that relate to choiceSetID_1 app sends
+-- - PerformInteraction request choiceSetID_1
+-- SDL does:
+-- - responde PerfromInteraction(resultCode: REJECTED, success: false) to app
```
Comment (dboltovskyi) [inline test_scripts/Defects/8_1/1883_2_ChoiceSet_being_processed_in_DeleteInteractionChoceSet_after_resumption.lua:17]: Typo `responde`

File: test_scripts/Defects/8_1/1883_1_ChoiceSet_being_processed_in_DeleteInteractionChoceSet.lua [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+-- Preconditions:
+-- 1. SDL and HMI are started
+-- 2. Mobile app is registered and activated
+-- In case:
+-- 1. App is created InteractionChoiseSet(choiceSetID_1)
+-- 2. App starts deleted InteractionChoiseSet(choiceSetID_1)
+-- - During HMI processes VR.DeleteCommands_request that relate to choiceSetID_1 app sends
+-- - PerformInteraction request choiceSetID_1
+-- SDL does:
+-- - respond PerfromInteraction(resultCode: REJECTED, success: false) to app
```
Comment (dboltovskyi) [inline test_scripts/Defects/8_1/1883_1_ChoiceSet_being_processed_in_DeleteInteractionChoceSet.lua:15]: I would rewrite description in a following way:
```
...
2. App sends DeleteInteractionChoiceSet(choiceSetID_1) request
SDL does transfer request to HMI
3. App sends PerformInteraction(choiceSetID_1) request
SDL does:
 - not transfer request to HMI
 - responds with REJECTED to the App

4. HMI responds to DeleteInteractionChoiceSet
SDL does: 
...
```

File: test_scripts/Defects/8_1/1883_2_ChoiceSet_being_processed_in_DeleteInteractionChoceSet_after_resumption.lua [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+---------------------------------------------------------------------------------------------------
+--[[ Required Shared libraries ]]
+local runner = require("user_modules/script_runner")
+local common = require("user_modules/sequences/actions")
+local utils = require("user_modules/utils")
+local SDL = require("SDL")
```
Comment (dboltovskyi) [inline test_scripts/Defects/8_1/1883_2_ChoiceSet_being_processed_in_DeleteInteractionChoceSet_after_resumption.lua:23]: Do we need `SDL` module?

File: test_scripts/Defects/8_1/1883_2_ChoiceSet_being_processed_in_DeleteInteractionChoceSet_after_resumption.lua [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+  :Do(function(_, data)
+      common.hmi.getConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+end
+
+local function unexpectedDisconnect()
+  common.getHMIConnection():ExpectNotification("BasicCommunication.OnAppUnregistered", { unexpectedDisconnect = true })
+  :Times(common.mobile.getAppsCount())
+  common.mobile.disconnect()
+  utils.wait(1000)
```
Comment (dboltovskyi) [inline test_scripts/Defects/8_1/1883_2_ChoiceSet_being_processed_in_DeleteInteractionChoceSet_after_resumption.lua:111]: `utils.wait` can be replaced by `common.run.wait`

File: test_scripts/Defects/8_1/1883_1_ChoiceSet_being_processed_in_DeleteInteractionChoceSet.lua (hunk 3 above)
Comment (dboltovskyi) [reply test_scripts/Defects/8_1/1883_1_ChoiceSet_being_processed_in_DeleteInteractionChoceSet.lua:15]: Same can be done for 2nd script

File: test_scripts/Defects/8_1/1883_1_ChoiceSet_being_processed_in_DeleteInteractionChoceSet.lua (hunk 1 above)
Comment (dboltovskyi) [reply test_scripts/Defects/8_1/1883_1_ChoiceSet_being_processed_in_DeleteInteractionChoceSet.lua:4]: Same can be done for 2nd script

File: test_sets/Defects/Defects_release_8_1.txt [hunk 6]
```diff
@@ -1 +1,3 @@
+./test_scripts/Defects/8_1/1883_1_ChoiceSet_being_processed_in_DeleteInteractionChoceSet.lua
+./test_scripts/Defects/8_1/1883_2_ChoiceSet_being_processed_in_DeleteInteractionChoceSet_after_resumption.lua
```
Comment (dboltovskyi) [inline test_sets/Defects/Defects_release_8_1.txt:2]: I would rename scripts in way of requirement. E.g.:
> ..._PerformInteraction_REJECTED_for_being_deleted_InteractionChoceSet

**Review by dboltovskyi — CHANGES_REQUESTED** [review]

File: test_scripts/Defects/8_1/1883_2_ChoiceSet_being_processed_in_DeleteInteractionChoceSet_after_resumption.lua (hunk 5 above)
Comment (VNikiforov-dev) [reply test_scripts/Defects/8_1/1883_2_ChoiceSet_being_processed_in_DeleteInteractionChoceSet_after_resumption.lua]: Done

File: test_sets/Defects/Defects_release_8_1.txt (hunk 6 above)
Comment (VNikiforov-dev) [reply test_sets/Defects/Defects_release_8_1.txt]: Done

File: test_scripts/Defects/8_1/1883_2_ChoiceSet_being_processed_in_DeleteInteractionChoceSet_after_resumption.lua (hunk 4 above)
Comment (VNikiforov-dev) [reply test_scripts/Defects/8_1/1883_2_ChoiceSet_being_processed_in_DeleteInteractionChoceSet_after_resumption.lua]: Extra SDL and utils. Done

**Comment by VNikiforov-dev** [comment]
@dboltovskyi Please find updates in 51a4d38

**Review by dboltovskyi — APPROVED** [review]
@VNikiforov-dev  Thank you for updates

File: test_scripts/Defects/8_1/1883_1_PerformInteraction_REJECTED_for_being_deleted_InteractionChoceSet.lua [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+-- 2. Mobile app is registered and activated
+-- In case:
+-- 1. App is created InteractionChoiseSet(choiceSetID_1)
+-- 2. App sends DeleteInteractionChoiceSet(choiceSetID_1) request
+-- SDL does:
+-- - transfer request to HMI
+-- 3. App sends PerformInteraction(choiceSetID_1) request
+-- SDL does:
+-- - not transfer request to HMI
+-- - responds with REJECTED to the App
```
Comment (VjKlepikov) [inline test_scripts/Defects/8_1/1883_1_PerformInteraction_REJECTED_for_being_deleted_InteractionChoceSet.lua:17]:
```suggestion
-- - respond with REJECTED to the App
```

File: test_scripts/Defects/8_1/1883_1_PerformInteraction_REJECTED_for_being_deleted_InteractionChoceSet.lua [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+-- 4. HMI responds to DeleteInteractionChoiceSet
+-- SDL does:
+-- - responds DeleteInteractionChoiceSet(SUCCESS) to App
```
Comment (VjKlepikov) [inline test_scripts/Defects/8_1/1883_1_PerformInteraction_REJECTED_for_being_deleted_InteractionChoceSet.lua:20]:
```suggestion
-- - respond DeleteInteractionChoiceSet(SUCCESS) to App
```
Check for all scripts

File: test_scripts/Defects/8_1/1883_2_PerformInteraction_REJECTED_for_being_deleted_InteractionChoceSet_after_resumption.lua [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+      sendPerformInteraction_REJECTED()
+      local function deleteCommandResp()
+        common.hmi.getConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+      end
+      common.run.runAfter(deleteCommandResp, 1000)
+    end)
+  common.mobile.getSession():ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+end
+
+local function reRegisterApp(pParams)
```
Comment (VjKlepikov) [inline test_scripts/Defects/8_1/1883_2_PerformInteraction_REJECTED_for_being_deleted_InteractionChoceSet_after_resumption.lua:104]: Please remove extra variable pParams
```suggestion
local function reRegisterApp()
```

**Review by VjKlepikov — APPROVED** [review]

File: test_scripts/Defects/8_1/1883_2_PerformInteraction_REJECTED_for_being_deleted_InteractionChoceSet_after_resumption.lua (hunk 9 above)
Comment (VNikiforov-dev) [reply test_scripts/Defects/8_1/1883_2_PerformInteraction_REJECTED_for_being_deleted_InteractionChoceSet_after_resumption.lua]: Done

File: test_scripts/Defects/8_1/1883_1_PerformInteraction_REJECTED_for_being_deleted_InteractionChoceSet.lua (hunk 8 above)
Comment (VNikiforov-dev) [reply test_scripts/Defects/8_1/1883_1_PerformInteraction_REJECTED_for_being_deleted_InteractionChoceSet.lua]: Done

**Comment by dboltovskyi** [comment]
Moved to `https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2611`

**Pull request closed without merging**